internal/controller/drplacementcontrol_controller.go

```diff
@@ -166,13 +182,21 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
 		err := r.processDeletion(ctx, drpc, placementObj, logger)
 		if err != nil {
-			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
-
 			statusErr := r.setDeletionStatusAndUpdate(ctx, drpc)
 			if statusErr != nil {
 				err = fmt.Errorf("drpc deletion failed: %w and status update failed: %w", err, statusErr)
+
+				return ctrl.Result{}, err
+			}
+
+			// Is this an expected condition?
+			if errorswrapper.Is(err, DeleteInProgress{}) {
+				logger.Info("Deleting DRPC in progress", "reason", err)
+
+				return ctrl.Result{RequeueAfter: DeleteInProgressDelay}, nil
```

Comment:
This is likely to break the unit tests that try to test entire flow in less then 1 second. Need to check how other delays are handled in the tests.